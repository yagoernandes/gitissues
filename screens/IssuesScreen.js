import React, { useState, useEffect } from 'react'
import {
  ScrollView,
  StyleSheet,
  View,
  Text,
  TouchableOpacity,
  Image
} from 'react-native'
import { Ionicons } from '@expo/vector-icons'
import axios from 'axios'
import * as WebBrowser from 'expo-web-browser'


export default function IssuesScreen({
  navigation: {
    state: {
      params: { avatar, id, login, name }
    }
  }
}) {
  const [state, setState] = useState({
    filter: 'all',
    issues: []
  })

  useEffect(() => {
    axios
      .get(
        `http://api.github.com/repos/${name}/${login}/issues?state=${state.filter}`
      )
      .then(response => {
        if (response.data && response.data.id) {
          console.log('Resposta: ', response.data)
          // console.log('ID: ', response.data.id)
          // console.log('Título: ', response.data.title)
          // console.log('Usuário: ', response.data.user.login)
          // console.log('Avatar: ', response.data.user.avatar_url)

          setState(s => ({ ...s, issues: response.data }))
        }
      })
  }, [state.filter])

  return (
    <View style={styles.container}>
      <View style={styles.topBar}>
        <TouchableOpacity onPress={()=>{setState(s=>({...s, filter:'all'}))}}>
          <Text style={styles.topBarTextSelected}>Todas</Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={()=>{setState(s=>({...s, filter:'open'}))}}>
          <Text style={styles.topBarText}>Abertas</Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={()=>{setState(s=>({...s, filter:'closed'}))}}>
          <Text style={styles.topBarText}>Fechadas</Text>
        </TouchableOpacity>
      </View>
      <ScrollView style={styles.list}>
        {state.issues.map(issue => (
          <Issue issue={issue} />
        ))}
      </ScrollView>
    </View>
  )
}

IssuesScreen.navigationOptions = {
  header: null
}

function Issue({ issue }) {
  return (
    <TouchableOpacity
      onPress={() => {
        WebBrowser.openBrowserAsync(issue.url)
      }}>
      <View style={styles.item}>
        <View style={styles.avatar}>
          <Image source={{ uri: issue.user.avatar_url }} style={styles.image} />
        </View>
        <View style={styles.repoText}>
          <Text style={styles.repoTitle}>{issue.title}</Text>
          <Text style={styles.repoSubtitle}>{issue.user.login}</Text>
        </View>
        <View style={styles.itemIcon}>
          <Ionicons name='ios-arrow-forward' size={32} color='#CCC' />
        </View>
      </View>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#EEE',
    paddingHorizontal: 20,
    paddingVertical: 15
  },
  list: {
    paddingTop: 15
  },
  topBar: {
    backgroundColor: '#CCC',
    borderRadius: 5,
    display: 'flex',
    flexDirection: 'row',
    paddingVertical: 10,
    paddingHorizontal: 20,
    justifyContent: 'space-between'
  },
  topBarText: {
    color: '#AAA'
  },
  topBarTextSelected: {
    color: '#666',
    fontWeight: '700'
  },
  item: {
    display: 'flex',
    flexDirection: 'row',
    backgroundColor: '#FFF',
    borderRadius: 5,
    padding: 10,
    marginBottom: 15
  },
  image: {
    width: 60,
    height: 60,
    resizeMode: 'contain',
    borderRadius: 100,
    marginRight: 15
  },
  repoText: {
    flex: 1,
    justifyContent: 'center'
  },
  repoTitle: {
    fontWeight: '700',
    fontSize: 18,
    marginBottom: 3
  },
  repoSubtitle: {
    color: '#CCC',
    fontSize: 14
  },
  itemIcon: {
    paddingRight: 10,
    justifyContent: 'center'
  }
})
