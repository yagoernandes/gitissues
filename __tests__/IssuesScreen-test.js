import React from 'react'
import NavigationTestUtils from 'react-navigation/NavigationTestUtils'
import renderer from 'react-test-renderer'
// import { render, fireEvent } from 'react-testing-library'
import IssuesScreen from '../screens/IssuesScreen'

describe('IssuesScreen', () => {
  jest.useFakeTimers()

  it(`renders the issues screen`, () => {
    const fakeState = {
      state: {
        params: { id: 123123123, name: 'Test', login: 'testtt', avatar: 'url' }
      }
    }
    const tree = renderer
      .create(<IssuesScreen navigation={fakeState} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
