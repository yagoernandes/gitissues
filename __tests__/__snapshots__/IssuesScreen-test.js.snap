// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssuesScreen renders the issues screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#EEE",
      "flex": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 15,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#CCC",
        "borderRadius": 5,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#666",
            "fontWeight": "700",
          }
        }
      >
        Todas
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#AAA",
          }
        }
      >
        Abertas
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#AAA",
          }
        }
      >
        Fechadas
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "paddingTop": 15,
        "textAlign": "center",
      }
    }
  >
    Carregando...
  </Text>
</View>
`;
